### PR TITLE
Add tooltip for toggling favorite nodes

### LINKED ISSUE #84564: `Switch to Favorite Nodes` button tooltip doesn't change
### Godot version

v4.2.beta1.official [b1371806a]

### System information

Godot v4.1.1.stable - Windows 10.0.19045 - Vulkan (Mobile) - integrated AMD Radeon(TM) Vega 8 Graphics (Advanced Micro Devices, Inc.; 31.0.21029.1006) - AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx (8 Threads)

### Issue description

image
image

The tooltip should change depending on if it's pressed or not.

Pseudocode:
```
if not pressed:
    tooltip = "Switch to Favorite Nodes"
else:
    tooltip = "Switch to Default Nodes"
```

### Steps to reproduce

1. Click `Scenee > New Scene`;

2. Locate this window: 

3. Hover over "star" button;

### Minimal reproduction project

Present in any new project.

### PR DESCRIPTION
**Pull request by quirkylemon**
fixes #84564 
This needs translations

### CODE REVIEW

**Comment by KoBeWi** [comment]
Wouldn't it be better to change the tooltip to e.g. "Toggle display favorite nodes.", so that it's always relevant?

**Comment by AThousandShips** [comment]
Agreed, that is also on the above line 1367

**Comment by quirkylemon** [comment]
> Wouldn't it be better to change the tooltip to e.g. "Toggle display favorite nodes.", so that it's always relevant?

Yeah that makes a lot more sense

**Review by KoBeWi — APPROVED** [review]

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -1364,7 +1364,7 @@ void SceneTreeDock::_notification(int p_what) {
 			node_shortcuts_toggle->set_flat(true);
 			node_shortcuts_toggle->set_icon(get_editor_theme_icon(SNAME("Favorites")));
 			node_shortcuts_toggle->set_toggle_mode(true);
-			node_shortcuts_toggle->set_tooltip_text(TTR("Switch to Favorite Nodes"));
+			node_shortcuts_toggle->set_tooltip_text(TTR("Toggle display favorite nodes."));
```
Comment (akien-mga) [inline editor/scene_tree_dock.cpp:1367]:
```suggestion
			node_shortcuts_toggle->set_tooltip_text(TTR("Toggle the display of favorite nodes."));
```

Comment (quirkylemon) [reply editor/scene_tree_dock.cpp]: Sorry it took so long to get to this.

**Review by Calinou — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**